activerecord/lib/active_record/associations/belongs_to_association.rb

```diff
@@ -77,6 +77,8 @@ def replace(record)
             raise_on_type_mismatch!(record)
             set_inverse_instance(record)
             @updated = true
+          elsif inverse_on_target = inverse_association_for(target)
+            inverse_on_target.reset
```

Comment:
Hmm, ok the build is failing because we still need to check if `target` isn't `nil`.

Looking at association I see there is a `remove_inverse_instance` we can use.
https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations/association.rb#L131-L136

```suggestion
          elsif target
            remove_inverse_instance(target)
```